src/mrfgen/mrfgen.py

```diff
@@ -562,13 +668,22 @@ def run_mrf_insert(mrf, tiles, insert_method, resize_resampling, target_x, targe
     log_info_mssg("Inserting new tiles into " + mrf)
     mrf_insert_command_list = ['mrf_insert', '-r', insert_method]
 
+    should_lock = parallel and merge
+
+    if should_lock:
+        global lock
+
     for tile in tiles:
+        if should_lock:
```

Comment:
What read error are we aiming to prevent in this block?

Reply:
The locking is in place to prevent gdal_merge calls from occurring at the same time as mrf_insert. This leads to overlapping and empty tiles for products like MODIS_Chlorophyll_A where the image granules overlap frequently. The rw_lock ensures mutual exclusion between merge and insert.